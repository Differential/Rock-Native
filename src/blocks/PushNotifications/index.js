//@flow

import pushNotifications from "./pushNotifications";

export default pushNotifications;
