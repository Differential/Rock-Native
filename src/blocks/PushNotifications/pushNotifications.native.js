// @flow

import React, { Component } from "react";
import { Button } from "react-native";
import FCM from "react-native-fcm";
import { lifecycle } from "recompose";
// import FCM, {FCMEvent, RemoteNotificationResult, WillPresentNotificationResult, NotificationType} from 'react-native-fcm';
// import { withState, lifecycle } from "recompose";

export const withLifeCycle = (FCM) => lifecycle({
  componentDidMount() {
    // From initial notification probably want some router handlers?
    FCM.getInitialNotification().then(notif => {
      console.log("INITIAL NOTIFICATION", notif); // eslint-disable-line
    });
  }
});

export const requestPermission = (FCM) => () => {
  FCM.requestPermissions();
  FCM.getFCMToken().then(token => {
    console.log("TOKEN (getFCMToken)", token); // eslint-disable-line
  });
}

export const PushRequestButton = () =>
  <Button onPress={requestPermission(FCM)} title="Allow Push Notifications"/>

const PushRequestButtonWithLifecycle = withLifeCycle(FCM)(PushRequestButton)
PushRequestButtonWithLifecycle.displayName = "PushNotificationRequestButton";

export default PushRequestButtonWithLifecycle;

// Set Badge on App, only for iOS Make this a Button?
// get Badge number
