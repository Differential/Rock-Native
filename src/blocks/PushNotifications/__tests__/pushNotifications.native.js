import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import PushNotifications from "../pushNotifications.native.js";

// Mock calls to FCM.
// jest.mock("react-native-fcm");

const generateComponent = additionalProps => (
  <PushNotifications {...additionalProps} />
);

describe("Push Notifications", () => {
  it("should render the Push Notifications component for native", () => {
    const tree = shallow(generateComponent());
    expect(tree).toBeDefined();
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("should call FCM");
});
