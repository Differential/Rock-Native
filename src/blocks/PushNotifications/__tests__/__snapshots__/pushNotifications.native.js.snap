// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Exported Button should render the Push Notifications component for native 1`] = `
<Button
  onPress={[Function]}
  title="Allow Push Notifications"
/>
`;
